frontend/src/tests/ForestClientInput.spec.ts

```diff
@@ -75,31 +107,297 @@ describe('ForestClientInput', () => {
+    //     expect(wrapper.find("[aria-label='Add Client Numbers']").element.textContent).contains('Loading')
+    //     await flushPromises();
+
+    //     // expect(isLoading()).toBe(true);
+    // });
+
+    it('should test the delete forest client number case', async () => {
+        await input.setValue(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+
+        await wrapper.find("[aria-label='Add Client Numbers']").trigger('click');
+        await flushPromises();
+
+        expect(wrapper.findAll('#forest-client-id')).toHaveLength(1);
+
+        const deleteButton = wrapper.find('#btn-trash-can');
+        deleteButton.trigger('click');
+        await flushPromises();
+
+        expect(wrapper.findAll('#forest-client-id')).toHaveLength(0);
+
+        // removeVerifiedForestClients has been called when delete is clicked
+        expect(wrapper.emitted()).toHaveProperty('removeVerifiedForestClients');
+    });
+
+    it('should cleanup error when input change', async () => {
+        await input.setValue(TEST_INACTIVE_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toBe(TEST_INACTIVE_FOREST_CLIENT_NUMBER);
+
+        await wrapper.find("[aria-label='Add Client Numbers']").trigger('click');
+        await flushPromises();
+
+        // Check if the error message has been raised
+        expect(wrapper.find('#forestClientInputValidationError').element.textContent).toBeTruthy();
+
+        await input.setValue(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        // The error message should be cleared
+        expect(wrapper.find('#forestClientInputValidationError').exists()).toBeFalsy();
+    });
+
+    it('should cleanup forest client card when property change', async () => {
+        await input.setValue(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER);
+
+        await wrapper.find("[aria-label='Add Client Numbers']").trigger('click');
+        await flushPromises();
+
+        const clientIdCardList = wrapper.findAll('#forest-client-id');
+        expect(clientIdCardList).toHaveLength(1);
+
+        // Changing the userId prop should cleanUp Forest Client Card
+        await wrapper.setProps({ userId: 'Test' });
+
+        expect(wrapper.emitted()).toHaveProperty('resetVerifiedForestClients');
+
+        // The forest client card should be cleared
+        expect(wrapper.findAll('#forest-client-id')).toHaveLength(0);
+
+        await input.setValue(TEST_SUCCESS_FOREST_CLIENT_NUMBER_2);
+        expect(inputField.value).toBe(TEST_SUCCESS_FOREST_CLIENT_NUMBER_2);
+
+        await wrapper.find("[aria-label='Add Client Numbers']").trigger('click');
+        await flushPromises();
+
+        expect(wrapper.findAll('#forest-client-id')).toHaveLength(1);
+
+        await wrapper.setProps({ userId: 'Test2' });
```

Comment:
why we repeat the test when set props userId? do you want to set roleId here and verify the forest client card is removed?